### ISSUE TITLE
Cannot build project. At all.

### ISSUE THREAD

**Issue by FirEmerald** [AxioDL/urde#19]
here is my log: http://pastebin.com/qiRBnbkF
there are a LOT of errors here that shouldn't be there, can you please fix this?
Also, I wanted to build this project to get a better idea of how the thermal visor's pallet is generated, for my Minecraft Mod Metroid Cubed 3.

**Comment by Antidote** [comment]
Hmmm, looks like a string concatenation error, I don't have URDE on me atm, but next chance i get I'll look into it.

**Comment by FirEmerald** [comment]
Actually, some of the errors I see are a missing identifier - atStat64()

**Comment by Antidote** [comment]
Ah, yeah I see them, that's caused by a recent change made in Athena, I'll whip up a fix real quick and squash that error, however that's definitely not the only one. The issue is that neither Jack nor myself are active Windows users so the Windows side of things tends to get neglected.

**Comment by jackoalan** [comment]
Ok, I've addressed some Windows inconsistencies that came up over the last several commits.
Please pull the latest and update submodules recursively to get everything up to date:

**Comment by jackoalan** [comment]
Oh, if you're looking to get a fully integrated thermal visor rendering, just know that URDE is very much a work-in-progress, and there isn't currently a way to get it rendering without directly modifying the C++.

However, here you'll find a video and technical description of the Thermal Visor:
https://m2k2.taigaforum.com/post/urde_43.html#urde_43

**Comment by jackoalan** [comment]
TXTR_ThermoPalette.png (and all other game resources) may be extracted from a disc image of Metroid Prime (NTSC/PAL/Trilogy) using hecl.exe built from this repository (`hecl extract GM8E01.gcm`)

Please note that this is a palette-based texture (specifically a C4 texture in the GX specification). Loading the colormap in GIMP looks like so:
![screen shot 2016-10-20 at 12 37 46](https://cloud.githubusercontent.com/assets/505536/19580466/1f1f4298-96c2-11e6-92f0-0336fa5047cb.png)

**Comment by FirEmerald** [comment]
Posting this because email-reply seems to have not worked.

I have the black-and-white shader done, I just needed to know how to generate the color pallet. I did extract all the files, but there was no TXTR_ThermalPallete.png extracted. I used trilogy, but I also have the original GC version as well. Also, the last release (v9) won't extract MP3 or MP2 without crashing, and struggled to extract MP1 without crashing.

Also there are 2 more errors that need adressing, then it SHOULD be working:
4>clangDriver.lib(MSVCToolChain.obj) : error LNK2001: unresolved external symbol GetFileVersionInfoSizeW
4>clangDriver.lib(MSVCToolChain.obj) : error LNK2001: unresolved external symbol GetFileVersionInfoW

**Comment by jackoalan** [comment]
It's been awhile since i've tested MP2/3, most of our efforts are toward getting the details of GCN MP1 (specifically 0-00) figured out. I'll do an extract test next.

Which version of Windows are you using? Are you building with Visual Studio 2015? The missing GetFileVersionInfoW is somewhat unusual, since i've been able to get it linking on my Windows 10 machine. I'm not presently able to verify Windows 7/8 builds unfortunately.

**Comment by FirEmerald** [comment]
I'm using Windows 10, Visual Studio 2015 64-bit, I can try a rebuild but it takes almost an hour to do, there's so much code. what is the purpose of GetFileVersionInfoSizeW and GetFileVersionInfoW in this code anyway? Is this even in your code? It looks like it might actually be a VS error (I'll try my OTHER visual Studio 2015, the x86 version.)

**Comment by Antidote** [comment]
LLVM uses it to embed the build date in the PE-COFF binaries, unfortunately there isn't much we can do, since it's used by `clang-tooling` which we use in `atdna`.

**Comment by jackoalan** [comment]
Those two functions part of Windows itself; has been since Windows 2000:
https://msdn.microsoft.com/en-us/library/windows/desktop/ms647003(v=vs.85).aspx

That's why i find it unusual. The error shouldn't be happening as long as `Mincore` is linked, and I know for sure our build system does that for Windows builds.

**Comment by FirEmerald** [comment]
I'll bing it.

**Comment by jackoalan** [comment]
Oh, are you trying to build a 32-bit binary of the codebase? That LLVM package is 64-bit only; you can't directly mix 32/64 bit code on *any* x86 OS.

We've decided to follow in Dolphin's footsteps and only officially support 64-bit, mainly because none of us run systems to test and debug 32-bit code. x86_64 has a far better ABI design anyway; more registers, so more optimal code can be generated. It's not *just* about more than 4GB of memory access.

**Comment by FirEmerald** [comment]
No, I was using the 64-bit build configuration, it's just the visual studio program itself was(IS) 32-bit. even after uninstalling the VS-2015 x86 and installing the x64 one, it says 32-bit in task manager. Go figure.
I know 64-bit is superior to 32-bit.

**Comment by jackoalan** [comment]
Gotcha, my misunderstanding

I'll do some more thorough testing on Windows, I've already found suggestions about using `version.lib` instead of `mincore.lib`. I'll try it out, and if it doesn't break here, I'll have you try building on your system

**Comment by FirEmerald** [comment]
fixed it - add "version.lib" to atdna's required dependencies.

**Comment by jackoalan** [comment]
sweet! thank you for testing that!!

i'll be sure that gets into the link libs in the next commit

**Comment by FirEmerald** [comment]
I have the black-and-white shader done, I just needed to know how to
generate the color pallet. I did extract all the files, but there was no
TXTR_ThermalPallete.png extracted. I used trilogy, but I also have the
original GC version as well. Also, the last release (v9) won't extract MP3
or MP2 without crashing, and struggled to extract MP1 without crashing.

Also there are 2 more errors that need adressing, then it SHOULD be working:
4>clangDriver.lib(MSVCToolChain.obj) : error LNK2001: unresolved external
symbol GetFileVersionInfoSizeW
4>clangDriver.lib(MSVCToolChain.obj) : error LNK2001: unresolved external
symbol GetFileVersionInfoW
